GUIDE.md

```diff
@@ -1,28 +1,217 @@
-# Server Endpoint
+# Guide
 
+ - The guiding focus to how to write a client to connect to a Tokyo server, receive information events and create actions to control your ship
+
+## 1. Connection information
+
+`ws://${host}/socket?key={key}&name{name}`
+
+| Parameters | Description |
+|--|--|
+| {host} | Domain or IP of server want to connect |
+| {key} | User's identity is unique and used to distinguish bots |
+| {name} | The display name will be shown on UI |
+
+When the WebSocket connection is established successfully, your bot is registered, displayed on web UI and ready to use.
+
+## 2. Events
+
+From WebSocket, the server consecutively sends events to the client every tick with the following structure.
+Event message structure contains all world info of the game, including players, bullets, dead person, etc.
+
+### 2.1. Events structure
+
+#### 2.1.1. State event
+
+State event contains world map states that includes map info, players, bullets, dead and scoreboard
+
+```json
+{
+  "e": "state",
+  "data": {
+    "bounds": [1200.0, 800.0],
+    "players": [
+      {},
```

Comment:
This `{}` really doesn't make sense....would it be nicer if we write this as a typescript interface rather than raw json. :thinking:

Reply:
But this is nitpicking so you don't have to.

Reply:
But I don't know typescript, then let's skip it